### PR TITLE
Modal ask user to login

### PR DESCRIPTION
**Pull request by akiyama-eve**
- Modal ask user to go to login before upload image or submit article
- Fix modal bug
screen shot 2560-08-23 at 11 35 21 am

### CODE REVIEW

File: src/app/containers/Modal/containers/ModalDispatch.js [hunk 1]
```diff
@@ -10,17 +10,18 @@ import { fetchCard, setCard } from "../../../../actions/modalActions"
 class ModalDispatch extends React.Component {
   invokeModal = event => {
     event.preventDefault()
-    this.props.with.request.history = this.props.history
-    this.props.with.request ?
+    if(this.props.with.request){
+      this.props.with.request.history = this.props.history
```
Comment (dmitrizzle) [inline src/app/containers/Modal/containers/ModalDispatch.js]: What does `request.history` and `this.props.history` do?

File: src/app/components/_screens/Composer.js [hunk 2]
```diff
@@ -17,9 +17,9 @@ var requestData = {
```
Comment (dmitrizzle) [inline src/app/components/_screens/Composer.js]: Please see https://www.analog.cafe/my-stuff for an example of a full-screen login prompt

File: src/app/containers/Composer/containers/ContentEditor/index.js [hunk 3]
```diff
@@ -6,6 +6,8 @@ import { withRouter } from "react-router"
 
 // components
 import ImageButton from "./components/ImageButton"
+import { ModalDispatch } from "../../../../containers/Modal"
+import { fetchCard, setCard } from "../../../../../actions/modalActions"
```
Comment (dmitrizzle) [inline src/app/containers/Composer/containers/ContentEditor/index.js]: I don't think we need
```
import { ModalDispatch } from "../../../../containers/Modal"
import { fetchCard, setCard } from "../../../../../actions/modalActions"
```
imports anymore

**Comment by akiyama-eve** [comment]
To confirm my understand
1. When user click "Send Submission" will only display consent modal with not submit anything yet.
2. If user click "Yes" in consent modal will submit article with set all images with "full-consent".
3. If user click "No" will submit and set images with "basic-consent".
4. Do nothing if user close consent modal or refresh page.

Check user logged in, upload images and submit article when they click Yes or No in consent modal.

**Comment by dmitrizzle** [comment]
> When user click "Send Submission" will only display consent modal with not submit anything yet.

Yes, that's correct.

> If user click "Yes" in consent modal will submit article with set all images with "full-consent".

It's not the modal that will do the submission. It will be **another component** which is a full-page view and will appear on this url: `/submit/confirm-full-consent`. I haven't made that component yet though; will add it within the next hour.

> If user click "No" will submit and set images with "basic-consent".

Same as above, the route will point to `/submit/confirm-basic-consent` and the above component will take should take care of it. The reason for this is so that if the user needs to log-in, they could be simply redirected to one of the routes above to seamlessly continue submission.

> Do nothing if user close consent modal or refresh page.

Yes, that's correct.

> Check user logged in, upload images and submit article when they click Yes or No in consent modal.

See above; this process will happen within the component that responds to the route.

**Comment by dmitrizzle** [comment]
@akiyama-eve Please rebase and see `/src/app/containers/_screens-auth/SubmitConfirm.js` - this is the component that should take care of uploading user submission and checking whether they are logged in (not the card pop-up):

1. User chooses their consent level:
screen shot 2017-08-19 at 4 02 33 pm

2. They are directed to upload page that either asks them to log in or proceeds to upload submission:
screen shot 2017-08-23 at 2 43 50 pm

**Pull request closed without merging**